source/EngineTests/GenomeDescriptionEditServiceTests.cpp

```diff
@@ -756,3 +756,408 @@ TEST_F(GenomeDescriptionEditServiceTests, createSubGenomesForPreview_trimming_re
+}
+
+TEST_F(GenomeDescriptionEditServiceTests, createSeedCollectionForPreview_singleSubGenome_noCache)
+{
+    auto genome = GenomeDescription().genes({
+        GeneDescription().separation(false).nodes({
+            NodeDescription(),
+            NodeDescription(),
+        }),
+    });
+    
+    SubGenomeDescription subGenome{genome, 0, false};
+    std::vector<SubGenomeDescription> subGenomes = {subGenome};
+    std::unordered_map<SubGenomeDescription, Description> cache;
+    
+    auto result = GenomeDescriptionEditService::get().createSeedCollectionForPreview(subGenomes, cache);
+    
+    // Should have 1 seed creature
+    ASSERT_EQ(1, result.description._creatures.size());
+    ASSERT_EQ(1, result.seedCreatureIds.size());
+    
+    // Check creature properties
+    auto const& creature = result.description._creatures.at(0);
+    EXPECT_EQ(0, creature._generation);
+    EXPECT_EQ(1, creature._cells.size());
+    EXPECT_EQ(result.seedCreatureIds.at(0), creature._id);
+    
+    // Check cell position
+    auto const& cell = creature._cells.at(0);
+    EXPECT_FLOAT_EQ(toFloat(PREVIEW_HEIGHT) / 2, cell._pos.x);
+    EXPECT_FLOAT_EQ(toFloat(PREVIEW_HEIGHT) / 2, cell._pos.y);
+    
+    // Check genome reference
+    EXPECT_EQ(creature._genomeId, result.description._genomes.at(0)._id);
+}
+
+TEST_F(GenomeDescriptionEditServiceTests, createSeedCollectionForPreview_singleSubGenome_withCache)
+{
+    auto genome = GenomeDescription().genes({
+        GeneDescription().separation(false).nodes({
+            NodeDescription(),
+            NodeDescription(),
+        }),
+    });
+    
+    SubGenomeDescription subGenome{genome, 0, false};
+    
+    // Create cached phenotype with seed (generation 0) and offspring (generation 1)
+    Description cachedPhenotype;
+    cachedPhenotype._genomes.emplace_back(genome);
+    cachedPhenotype._creatures.emplace_back(
+        CreatureDescription()
+            .generation(0)
+            .genomeId(genome._id)
+            .cells({CellDescription().pos(RealVector2D{0, 0})})
+    );
+    cachedPhenotype._creatures.emplace_back(
+        CreatureDescription()
+            .generation(1)
+            .genomeId(genome._id)
+            .ancestorId(cachedPhenotype._creatures.at(0)._id)
+            .cells({CellDescription().pos(RealVector2D{1, 1})})
+    );
+    
+    std::vector<SubGenomeDescription> subGenomes = {subGenome};
+    std::unordered_map<SubGenomeDescription, Description> cache;
+    cache[subGenome] = cachedPhenotype;
+    
+    auto result = GenomeDescriptionEditService::get().createSeedCollectionForPreview(subGenomes, cache);
+    
+    // Should have both seed and offspring from cache
+    ASSERT_EQ(2, result.description._creatures.size());
+    ASSERT_EQ(1, result.seedCreatureIds.size());
+    
+    // Check that seed creature ID points to generation 0 creature
+    auto seedCreatureId = result.seedCreatureIds.at(0);
+    auto const& seedCreature = result.description._creatures.at(0);
```

Comment:
The test should not assume that the creature with generation = 0 is located at result.description._creatures.at(0). 
Instead test that both creatures: gen0 and gen1 are in the array result.description._creatures.